src/Ocelot/Requester/HttpClientWrapper.cs

```diff
@@ -12,13 +12,9 @@ public HttpClientWrapper(HttpClient client)
             Client = client;
         }
 
-        public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request,
-            CancellationToken cancellationToken = default)
+        public Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken = default)
         {
-            // https://www.stevejgordon.co.uk/using-httpcompletionoption-responseheadersread-to-improve-httpclient-performance-dotnet
-            // When using this option, we avoid the intermediate MemoryStream buffer, instead of getting the content directly from the stream exposed on the Socket.
-            // This avoids unnecessary allocations which is a goal in highly optimised situations.
-            return Client.SendAsync(request, HttpCompletionOption.ResponseHeadersRead, cancellationToken);
+            return Client.SendAsync(request, cancellationToken);
```

Comment:
Can we separate small and large content buffering scenarios?

My idea is
We can separate full buffering and streaming scenarios, then apply
- Small content, small sizes with [0 up to DefaultSizeOfFullBuffering] should use full buffering into the buffer
- Large content, aka unknown Content-Length aka streaming scenario should use only streaming `HttpCompletionOption.ResponseHeadersRead` option

Sounds good?

Reply:
Yes, but that's out of scope, I shouldn't have put thoses changes in my last PR. That's something we should discuss later in the message invoker PR.

Reply:
🆗 let's discuss this design idea when we start Dec'23...